### PR TITLE
Add zero spin optimizations to `KerrSchild`

### PR DESCRIPTION
**Pull request by nikwit**
This adds some optimizations to the KerrSchild solution if the spin is zero. The optimized tags are the ones needed by the CurvedScalarWave evolution system and dependend ones. 

~~depends on #4148~~

### CODE REVIEW

**Comment by nilsdeppe** [comment]
LGTM, please ping me once you've rebased and I'll approve :)

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsvu** [comment]
the macOS CI failure is fixed in #4170

**Pull request merged**
